File: sdk/php/src/VeyClient.php
<?php

namespace Vey;

use GuzzleHttp\Client;
use GuzzleHttp\Exception\GuzzleException;

class VeyClient
{
    private string $apiKey;
    private string $apiEndpoint;
    private Client $client;

    public function __construct(string $apiKey, string $apiEndpoint = 'https://api.vey.example')
    {
        $this->apiKey = $apiKey;
        $this->apiEndpoint = $apiEndpoint;
        $this->client = new Client([
            'base_uri' => $apiEndpoint,
            'headers' => [
                'Authorization' => 'Bearer ' . $apiKey,
                'Content-Type' => 'application/json',
            ],
        ]);
    }

    /**
     * Validate an address for a specific country.
     *
     * @param Address $address
     * @param string $countryCode
     * @return ValidationResult
     * @throws GuzzleException
     */
    public function validateAddress(Address $address, string $countryCode): ValidationResult
    {
        $response = $this->client->post('/validate', [
            'json' => [
                'address' => $address->toArray(),
                'countryCode' => $countryCode,
            ],
        ]);

        $data = json_decode($response->getBody()->getContents(), true);
        return ValidationResult::fromArray($data);
    }

    /**
     * Normalize an address to standard format.
     *
     * @param Address $address
     * @param string $countryCode
     * @return Address
     * @throws GuzzleException
     */
    public function normalizeAddress(Address $address, string $countryCode): Address
    {
        $response = $this->client->post('/normalize', [
            'json' => [
                'address' => $address->toArray(),
                'countryCode' => $countryCode,
            ],
        ]);

        $data = json_decode($response->getBody()->getContents(), true);
        return new Address($data);
    }

    /**
     * Encode address components into a PID.
     *
     * @param array $components
     * @return string
     */
    public function encodePID(array $components): string
    {
        $parts = [];
        
        foreach (['country', 'admin1', 'admin2', 'locality'] as $key) {
            if (isset($components[$key])) {
                $parts[] = $components[$key];
            }
        }
        
        return implode('-', $parts);
    }
}

File: sdk/flutter/lib/src/vey_client.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'models/address.dart';
import 'models/validation_result.dart';

class VeyClient {
  final String apiKey;
  final String apiEndpoint;

  VeyClient({
    required this.apiKey,
    this.apiEndpoint = 'https://api.vey.example',
  });

  /// Validate an address for a specific country
  Future<ValidationResult> validateAddress(
    Address address,
    String countryCode,
  ) async {
    final response = await http.post(
      Uri.parse('$apiEndpoint/validate'),
      headers: {
        'Content-Type': 'application/json',
        'Authorization': 'Bearer $apiKey',
      },
      body: jsonEncode({
        'address': address.toJson(),
        'countryCode': countryCode,
      }),
    );

    if (response.statusCode == 200) {
      return ValidationResult.fromJson(jsonDecode(response.body));
    } else {
      throw Exception('Failed to validate address');
    }
  }

  /// Normalize an address to standard format
  Future<Address> normalizeAddress(
    Address address,
    String countryCode,
  ) async {
    final response = await http.post(
      Uri.parse('$apiEndpoint/normalize'),
      headers: {
        'Content-Type': 'application/json',
        'Authorization': 'Bearer $apiKey',
      },
      body: jsonEncode({
        'address': address.toJson(),
        'countryCode': countryCode,
      }),
    );

    if (response.statusCode == 200) {
      return Address.fromJson(jsonDecode(response.body));
    } else {
      throw Exception('Failed to normalize address');
    }
  }

  /// Encode address components into a PID
  String encodePID(Map<String, dynamic> components) {
    // Implementation would call core library or API
    final parts = <String>[];
    
    if (components['country'] != null) parts.add(components['country']);
    if (components['admin1'] != null) parts.add(components['admin1']);
    if (components['admin2'] != null) parts.add(components['admin2']);
    if (components['locality'] != null) parts.add(components['locality']);
    
    return parts.join('-');
  }
}
